rubin_sim/maf/metrics/cumulativeMetric.py

```diff
@@ -0,0 +1,47 @@
+import numpy as np
+from .baseMetric import BaseMetric
+
+
+__all__ = ["CumulativeMetric"]
+
+
+class CumulativeMetric(BaseMetric):
+    """For plotting up the cumulative number of observations.
+    Expected to be used with a UniSlicer or UserPointSlicer with one point.
+
+    Parameters
+    ----------
+    bins : `np.array` (None)
+        The points to interpolate the cumulative number of observations to. If None,
+        then the range of the data is used.
+    """
+
+    def __init__(
+        self,
+        metricName="Cumulative",
+        time_col="observationStartMJD",
+        night_col="night",
+        bins=None,
+        **kwargs
+    ):
+        super().__init__(
+            col=[time_col, night_col],
+            metricName=metricName,
+            metricDtype="object",
+            **kwargs
+        )
+        self.time_col = time_col
+        self.night_col = night_col
+        self.bins = bins
+        self.plotDict = {"xlabel": "MJD (days)", "ylabel": "N obs"}
+
+    def run(self, dataSlice, slicePoint=None):
+        dataSlice.sort(order=self.time_col)
+        if self.bins is None:
```

Comment:
I don't understand why this works, so I must be missing something. 
If you're counting the number of visits in each night, I would have thought you could just use np.histogram. 
So then I tried both of these with a very stupid test case, and I don't get results that make sense to me. 
<img width="710" alt="Screen Shot 2022-02-01 at 2 14 23 PM" src="https://user-images.githubusercontent.com/1855711/152060443-210d0f2e-c68d-4093-9f6c-b58b9d8d7a41.png">

Reply:
Ah, `bins` is a terrible kwarg name. There's no actual binning happening. I just put the interpolation step in there to avoid plotting 2 million line segments. Updated to a better name.

Reply:
Ok, I see. And the other thing I was missing was a sort.